crates/primitives/src/snapshot/segment.rs

```diff
@@ -272,7 +297,7 @@ impl SegmentHeader {
     /// Returns the row offset which depends on whether the segment is block or transaction based.
     pub fn start(&self) -> u64 {
         match self.segment {
-            SnapshotSegment::Headers => self.block_start(),
+            SnapshotSegment::Headers => self.expected_block_start(),
             SnapshotSegment::Transactions | SnapshotSegment::Receipts => self.tx_start(),
```

Comment:
i see that `self.tx_start()` panics if `tx_range` is `None`. Shouldn't it be the same for the block range, i.e. we just take `self.block_start()` and panic on `None`?

Don't understand this part of code